Assets/Scripts/EOSSessionsManager.cs

```diff
@@ -1862,5 +2021,419 @@ public void DeclineLobbyInvite()
+                LocalUserId = EOSManager.Instance.GetProductUserId(),
+                SocketId = socketId
+            };
+
+            P2PSessionPeerRequestConnectionNotificationId = EOSManager.Instance.GetEOSPlatformInterface().GetP2PInterface().AddNotifyPeerConnectionRequest(ref options, null, OnIncomingSessionsConnectionRequest);
+
+            if (P2PSessionPeerRequestConnectionNotificationId == 0)
+            {
+                Debug.Log("EOSSessionsManager (SubscribeToSessionMessageConnectionRequests): Failed to subscribe to P2P Messages, bad Notification Id was returned.");
+            }
+
+            AddNotifyPeerConnectionClosedOptions closedOptions = new AddNotifyPeerConnectionClosedOptions()
+            {
+                LocalUserId = EOSManager.Instance.GetProductUserId(),
+                SocketId = socketId
+            };
+            P2PSessionPeerDisconnectConnectionNotificationId = EOSManager.Instance.GetEOSPlatformInterface().GetP2PInterface().AddNotifyPeerConnectionClosed(ref closedOptions, null, OnIncomingSessionsDisconnect);
+        }
+
+        /// <summary>
+        /// If subscribed, unsubscribe to P2P Session management messages.
+        /// The ability to receive P2P connection requests is unsubscribed from.
+        /// Closes all connections relating to the <see cref="P2PSessionStatusSocketName"/> socket.
+        /// </summary>
+        private void UnsubscribeToSessionMessageConnectionRequests()
+        {
+            if (P2PSessionPeerRequestConnectionNotificationId != 0)
+            {
+                EOSManager.Instance.GetEOSPlatformInterface().GetP2PInterface().RemoveNotifyPeerConnectionRequest(P2PSessionPeerRequestConnectionNotificationId);
+                P2PSessionPeerRequestConnectionNotificationId = 0;
+            }
+
+            if (P2PSessionPeerDisconnectConnectionNotificationId != 0)
+            {
+                EOSManager.Instance.GetEOSPlatformInterface().GetP2PInterface().RemoveNotifyPeerConnectionClosed(P2PSessionPeerDisconnectConnectionNotificationId);
+                P2PSessionPeerDisconnectConnectionNotificationId = 0;
+            }
+
+            CloseConnectionsOptions closeOptions = new CloseConnectionsOptions() { LocalUserId = EOSManager.Instance.GetProductUserId(), SocketId = new SocketId() { SocketName = P2PSessionStatusSocketName } };
+            EOSManager.Instance.GetEOSPlatformInterface().GetP2PInterface().CloseConnections(ref closeOptions);
+        }
+
+        /// <summary>
+        /// Whenever a user attempts to create a connection, this method handles their connection request.
+        /// By default, accept all incoming connections.
+        /// </summary>
+        /// <param name="data">Data, containing the product user id of the connecting request.</param>
+        private void OnIncomingSessionsConnectionRequest(ref OnIncomingConnectionRequestInfo data)
+        {
+            if (!(bool)data.SocketId?.SocketName.Equals(P2PSessionStatusSocketName))
```

Comment:
This is definitely confusing! I borrowed it from `EOSPeer2PeerManager`, which does it this way.

Changed over to the way you've suggested here. Probably wanna update those other places, too? Maybe when those get reworks.

Reply:
I would say update those when changes are made to those pieces of code; otherwise leave them alone to limit the scope of the changes in the PR to reflect a focused set of changes.